firebase-functions/functions/src/choiceUpdated.ts

```diff
@@ -0,0 +1,29 @@
+﻿import * as functions from 'firebase-functions';
+import * as admin from 'firebase-admin';
+
+// Dummy implementation for now. Will set some chosen thing just something will be visible
+export const choiceUpdated = functions.firestore.document('/players/{documentId}')
+  .onUpdate(async (change: any, context: any) => {
+    const playerId = context.params.documentId;
+    try {
+      const doc = await admin.firestore().collection('players').doc(playerId).get();
+      const roomId = doc.data()?.room;
+      console.log('The room in question ' + roomId);
+      const snapshot = await admin.firestore().collection('players').where('room', '==', roomId).get();
+      let areAllChoicesPresent = true;
+      let choice = '';
+      snapshot.forEach((player: any) => {
+        if (!player.data().choice) {
+          areAllChoicesPresent = false;
+          return;
+        }
+        choice = player.data().choice;
+      });
+
+      if (areAllChoicesPresent) {
+        await admin.firestore().collection('rooms').doc(roomId).update({ winner: choice });
```

Comment:
I know it is a dummy code, but we save the playerId instead of the choice :)

Reply:
Yes, allow me to keep it that way for now, it will be changed after :) You will not have to suffer this for long